recorder.go

```diff
@@ -74,24 +96,35 @@ func getHostName(rawSpan basictracer.RawSpan) string {
 }
 
 func getServiceName(rawSpan basictracer.RawSpan) string {
-	s := getStringTag(rawSpan, string(ext.Component))
-	if s == "" {
-		s = getStringTag(rawSpan, string(ext.PeerService))
-		if s == "" {
-			return sensor.serviceName
+	// ServiceName can be determined from multiple sources and has
+	// the following priority (preferred first):
+	//   1. If added to the span via the OT component tag
+	//   2. If added to the span via the OT http.url tag
+	//   3. Specified in the tracer instantiation via Service option
```

Comment:
Wouldn't it make sense to change the priority here and have explicitly set service name to be priority 1? The motivation is that component and http.url have different semantical meaning in [OT](https://github.com/opentracing/specification/blob/master/semantic_conventions.md). Also the same service might be producing a number of spans from different components and targeting different urls.